src/rgw/rgw_op.cc

```diff
@@ -6767,43 +6767,38 @@ void RGWDeleteMultiObj::handle_individual_object(const rgw_obj_key& o, optional_
 
 void RGWDeleteMultiObj::execute(optional_yield y)
 {
-  RGWMultiDelDelete *multi_delete;
-  RGWMultiDelXMLParser parser;
-  const uint32_t max_aio = std::max<uint32_t>(1, s->cct->_conf->rgw_multi_obj_del_max_aio);
-  auto group = ceph::async::spawn_throttle{y, max_aio};
-  char* buf;
-
-  buf = data.c_str();
+  const char* buf = data.c_str();
   if (!buf) {
     op_ret = -EINVAL;
-    goto error;
+    return;
   }
 
+  RGWMultiDelXMLParser parser;
   if (!parser.init()) {
     op_ret = -EINVAL;
-    goto error;
+    return;
   }
 
   if (!parser.parse(buf, data.length(), 1)) {
     op_ret = -EINVAL;
-    goto error;
+    return;
   }
 
-  multi_delete = static_cast<RGWMultiDelDelete *>(parser.find_first("Delete"));
+  auto multi_delete = static_cast<RGWMultiDelDelete *>(parser.find_first("Delete"));
   if (!multi_delete) {
     op_ret = -EINVAL;
-    goto error;
-  } else {
+    return;
+  }
+
 #define DELETE_MULTI_OBJ_MAX_NUM      1000
```

Comment:
nit: Since you're in the area could we make this a constant and not a `#define`?

Reply:
addressed in https://github.com/ceph/ceph/pull/57641. also added client-facing error messages